sdk/schema/plugin.go

```diff
@@ -21,6 +21,16 @@ type Plugin struct {
 	Executables []Executable
 }
 
+// Return the default credential for a plugin, or the first one in the list if it isn't set
+func (p Plugin) DefaultCredentialType() CredentialType {
```

Comment:
what if all credentials are set to be the default one? Should we even allow this? I would argue we would need to enforce that only one credential can be the default one, in `plugin.go` via the validation checks. 

The alternative would be to let the user set as many credentials as they want as default only to select the first one in order as the actual default one which could be confusing. The only scenario I could think of right now for which a plugin could have more than 1 default credential type defined, is because it also has more than 1 executable, and one of those executable has a certain default credential whereas another has a different default credential. However, in this case we could drop the `DefaultCredential` field altogether and solely rely on the order of specification for selecting the default credential for a certain executable. This would need to be documented. What do you think?

Reply:
I don't think there really is such thing as a "default credential". There are just certain credentials needed for certain commands.

If this data is used in `op` to determine the order to configure the credentials, then we can also get that from the order the credentials have in the `Executable` definition, and avoid the whole concept of "default credentials".

Reply:
I am fine with this!

Reply:
So that would also apply to the `DefaultCredential` field in the credential schema.